spec/components/custom_wizard/mapper_spec.rb

```diff
@@ -250,4 +276,135 @@
+          Incorrect
+        {%-endif-%}
+      LIQUID
+      mapper = create_template_mapper(template_params, user1)
+      result = mapper.interpolate(
+        template.dup,
+        template: true,
+        user: true,
+        wizard: true,
+        value: true
+      )
+      expect(result).to eq("Correct")
+    end
+
+    it "can access data passed to render method as variable" do
+      template = "{{step_1_field_1.size}}"
+      mapper = create_template_mapper(template_params, user1)
+      result = mapper.interpolate(
+        template.dup,
+        template: true,
+        user: true,
+        wizard: true,
+        value: true
+      )
+      expect(result).to eq(template_params["step_1_field_1"].size.to_s)
+    end
+
+    it "doesn't parse the template when template param is false" do
+      template = <<-LIQUID.strip
+        {{ "w{step_1_field_1}" | size}}
+      LIQUID
+      mapper = create_template_mapper(template_params, user1)
+      result = mapper.interpolate(
+        template.dup,
+        template: false,
+      )
+      expect(result).to eq(template)
+    end
+
+    context "custom filter: 'first_non_empty'" do
+      it "gives first non empty element from list" do
+        template = <<-LIQUID.strip
+          {%- assign entry = "" | first_non_empty: step_1_field_1, step_1_field_2, step_1_field_3 -%}
+          {{ entry }}
+        LIQUID
+        mapper = create_template_mapper(template_params_non_empty, user1)
+        result = mapper.interpolate(
+          template.dup,
+          template: true,
+          user: true,
+          wizard: true,
+          value: true
+        )
+        expect(result).to eq(template_params_non_empty["step_1_field_3"])
+      end
+
+      it "gives first non empty element from list  when multiple non empty values present" do
+        template = <<-LIQUID.strip
+          {%- assign entry = "" | first_non_empty: step_1_field_1, step_1_field_2, step_1_field_3 -%}
+          {{ entry }}
+        LIQUID
+        mapper = create_template_mapper(template_params_multiple_non_empty, user1)
+        result = mapper.interpolate(
+          template.dup,
+          template: true,
+          user: true,
+          wizard: true,
+          value: true
+        )
+        expect(result).to eq(template_params_multiple_non_empty["step_1_field_2"])
+      end
+
+      it "gives empty if all elements are empty" do
+        template = <<-LIQUID.strip
+          {%- assign entry = "" | first_non_empty: step_1_field_1, step_1_field_2, step_1_field_3 -%}
+          {%- if entry -%}
+            {{ entry }}
+          {%- endif -%}
+        LIQUID
+        mapper = create_template_mapper(template_params_empty, user1)
+        result = mapper.interpolate(
+          template.dup,
+          template: true,
+          user: true,
+          wizard: true,
+          value: true
+        )
+        expect(result).to eq("")
+      end
+    end
+  end
+end
+
+def create_template_mapper(data, user)
```

Comment:
@fzngagan If you look at other examples of method definitions in our specs, they're at the start of the file

Reply:
fixed